### PR TITLE
Add option to show icon labels.

### PR DESCRIPTION
**Pull request by tellthemachines**
## Description

Addresses #10524 and #15311
Adds an option in the Options modal to show labels for the editor UI icons.

No labels have been added yet; this is just the option toggle for the post editor. Further PRs will address adding the actual labels in each area where they are needed: the top toolbar and the block toolbar. 

This and the upcoming PRs for the labels will replace #15830 as that PR has too many conflicts to be salvageable.

## How has this been tested?

Tested in browser.

## Screenshots 

Screen Shot 2020-07-28 at 5 19 22 pm

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by tellthemachines** [comment]
As it stands, this option only applies to the post editor. But this is a setting that will also be useful in the site editor; is there a way we could make it apply in both places? It would be awkward to have an identical setting for each screen.

File: packages/edit-post/src/components/options-modal/index.js [hunk 1]
```diff
@@ -105,6 +106,9 @@ export function OptionsModal( { isModalActive, isViewable, closeModal } ) {
 				</PageAttributesCheck>
 			</Section>
 			<MetaBoxesSection title={ __( 'Advanced panels' ) } />
+			<Section title={ __( 'Icon Labels' ) }>
+				<ShowIconLabels label="Show Icon Labels" />
```
Comment (Soean) [inline packages/edit-post/src/components/options-modal/index.js:110]: This string should be translatable.

**Comment by paaljoachim** [comment]
The category inside Options "Icon Labels" could perhaps instead be named "Labels" as it would make it possible to later on add if needed additional label options inside this category.

**Comment by ZebulanStanphill** [comment]
I'd change the option from "Show Icon Labels" to "Show button text labels". (Note capitalization.)

**Comment by noisysocks** [comment]
Code looks great 👍, but I don't think we should add this option to the plugin _before_ we have anything that uses it? It would be confusing, I think, to see this setting, click it, and not see anything happen.

**Comment by tellthemachines** [comment]
> Code looks great 👍, but I don't think we should add this option to the plugin _before_ we have anything that uses it? It would be confusing, I think, to see this setting, click it, and not see anything happen.

I agree, and the PR that adds this to the toolbar is coming up Very Soon. The reason I did it this way is the original PR to add this functionality floundered because it was a huge changeset and conflicts kept popping up with every passing day, so I thought it might be easier to divide into even smaller chunks.

**Comment by tellthemachines** [comment]
Aaaand #24304 has now materialised with all the labels for the top toolbar 😄

File: packages/edit-post/src/components/options-modal/index.js (hunk 1 above)
Comment (jasmussen) [reply packages/edit-post/src/components/options-modal/index.js]: It should also be sentence case, right? 

Thank you for taking this on, by the way!

Comment (ZebulanStanphill) [reply packages/edit-post/src/components/options-modal/index.js]: The string has already been changed to "Show button text labels".

Comment (jasmussen) [reply packages/edit-post/src/components/options-modal/index.js]: Awesome. My mistake for giving a superficial review.

File: packages/edit-post/src/components/options-modal/index.js [hunk 2]
```diff
@@ -125,6 +129,8 @@ export default compose(
 	withDispatch( ( dispatch ) => {
 		return {
 			closeModal: () => dispatch( 'core/edit-post' ).closeModal(),
+			toggleIconLabels: () =>
+				dispatch( 'core/edit-post' ).toggleFeature( 'showIconLabels' ),
```
Comment (noisysocks) [inline packages/edit-post/src/components/options-modal/index.js:133]: Doesn't look like this is used anywhere?

Comment (tellthemachines) [reply packages/edit-post/src/components/options-modal/index.js]: Whoops, refactored and forgot to remove it 😅

**Review by draganescu — APPROVED** [review]
#24304 is approved now and this makes sense 👍

**Review by aristath — APPROVED** [review]

**Pull request merged**